### PR TITLE
Clarify, extend, fix typos in documentation

### PR DESCRIPTION
**Pull request by fsoikin**

### CODE REVIEW

File: src/Elmish/Boot.purs [hunk 1]
```diff
@@ -1,35 +1,54 @@
+module Elmish.Boot
```
Comment (fsoikin) [inline src/Elmish/Boot.purs:1]: The diff here is a bit confusing, but the change is that I moved the module comment to `BootRecord`. This way it shows up on re-exports too.

File: src/Elmish/Component.purs [hunk 2]
```diff
@@ -364,10 +344,10 @@ wrapWithLocalState' name mkDef =
...
--- | the other hand, we cannot create a fresh class on every render, because
--- | then React will see it as a new component every time, and will reset its
--- | state every time.
+-- | component to React, which makes it reuse state, which leads to breaking
+-- | type safety. On the other hand, we cannot create a fresh class on every
+-- | render, because then React will see it as a new component every time, and
+-- | will reset its state every time.
 -- |
 -- | This means that we need some way of figuring out whether it needs to be
 -- | logically "same" component or "different", but there is no way to get that
```
Comment (mcordova47) [inline src/Elmish/Component.purs:353]:
```suggestion
-- | logically the "same" component or "different", but there is no way to get that
```

File: src/Elmish/State.purs [hunk 3]
```diff
@@ -13,18 +13,23 @@ import Data.Maybe (Maybe(..), fromMaybe)
...
--- work with inline embedding of components.
+-- | This type represents a strategy of storing UI component state. The strategy
+-- | is a function that takes initial state and returns a monadic equivalent of
+-- | lens for manipulating the state.
+-- |
+-- | Currently there are two strategies:
+-- |
+-- |    * `dedicatedStorage` stores state in a dedicated mutable cell. This
+-- |      strategy is used in `Elmish.Component.construct`.
+-- |    * `localState` stores state locally on React component instance - i.e.
```
Comment (mcordova47) [inline src/Elmish/State.purs:24]:
```suggestion
-- |    * `localState` stores state locally on the React component instance - i.e.
```

**Review by mcordova47 — APPROVED** [review]

File: src/Elmish/Component.purs (hunk 2 above)
Comment (fsoikin) [reply src/Elmish/Component.purs:353]: Oh, thank you very much. Russian doesn't have these bloody articles, and you can probably tell that I still haven't developed an intuition for when and how to use them :-(

Comment (mcordova47) [reply src/Elmish/Component.purs]: No worries! To make things more complicated, things like documentation often omit them in English, too.

**Pull request merged**